### PR TITLE
[23] [Test] Login screen

### PR DESCRIPTION
**Pull request by markgravity**
https://github.com/markgravity/flutter-survey/issues/23

## What happened

Write tests for Login screen
 
## Proof Of Work

Screen Shot 2021-05-06 at 10 49 49

### CODE REVIEW

File: test/modules/login/login_router_test.dart [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      router = LoginRouterImpl();
+    });
+
+    describe("its replaceToHomeScreen() is called", () {
+      beforeEach(() {
+        router.replaceToHomeScreen(buildContext);
+      });
+
+      it("triggers navigator to push replacement to Home screen", () {
+        verify(navigatorState.pushReplacementNamed(HomeModule.routePath));
```
Comment (manh-t) [inline test/modules/login/login_router_test.dart:36]: missing `called(1)`

File: test/modules/login/login_router_test.dart [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      });
+    });
+
+    describe("its pushToForgotPasswordScreen() is called", () {
+      beforeEach(() {
+        router.pushToForgotPasswordScreen(buildContext);
+      });
+
+      it("triggers navigator to push to Forgot Password screen", () {
+        verify(navigatorState.pushNamed(ForgotPasswordModule.routePath));
```
Comment (manh-t) [inline test/modules/login/login_router_test.dart:46]: missing `called(1)`

**Review by manh-t — COMMENTED** [review]
looks good 💯

**Review by manh-t — APPROVED** [review]
LGTM 💯

**Review by sleepylee — APPROVED** [review]

**Pull request merged**